### PR TITLE
start adding registry

### PR DESCRIPTION
**Pull request by godber**
This PR adds the asset registry to support bundled assets and other things.

refs: https://github.com/terascope/teraslice/pull/2666

### CODE REVIEW

File: asset/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import FileBatcher from '../src/file_exporter/processor';
+import Schema from '../src/file_exporter/schema';
+
+import FileFetcher from '../src/file_reader/fetcher';
+import FileSlicerOperation from '../src/file_reader/slicer';
+import Schema from '../src/file_reader/schema';
```
Comment (godber) [inline asset/src/index.ts:6]: The `Schema` objects in both the `file_exporter` and `file_reader` are named `Schema` and don't currently have unique names.  Can I rename them using `as` or something?  Or will this somehow result in trouble?

Comment (godber) [reply asset/src/index.ts]: I talked with Peter about this, I've finished this with his recommendations.

**Comment by godber** [comment]
I think the most likely way for me to screw this up is omitting or transposing something ... I think most of the transposition errors would be caught by my editor, complete omissions would not likely be detected.

The resulting zipfile was remarkably small, just 380kb.  It contained some wasm and lz4 strings ... like those things were included instead of omitted.

I guess it worked.

**Comment by godber** [comment]
Here's the file asset bundle being required

```
./asset-require.sh ./file-v2.0.2-node-12-bundle.zip
Archive:  ./file-v2.0.2-node-12-bundle.zip
  inflating: ./tmp.Yx5i1FJM/index.js
  inflating: ./tmp.Yx5i1FJM/asset.json
{
  file_exporter: { Processor: { default: [Getter] }, Schema: { default: [Getter] } },
  file_reader: {
    Fetcher: { default: [Getter] },
    Schema: { default: [Getter] },
    Slicer: { default: [Getter] }
  },
  file_reader_api: { API: { default: [Getter] }, Schema: { default: [Getter] } },
  file_sender_api: { API: { default: [Getter] }, Schema: { default: [Getter] } },
  hdfs_append: { Processor: { default: [Getter] }, Schema: { default: [Getter] } },
  hdfs_reader: {
    Fetcher: { default: [Getter] },
    Schema: { default: [Getter] },
    Slicer: { default: [Getter] }
  },
  hdfs_reader_api: { API: { default: [Getter] }, Schema: { default: [Getter] } },
  hdfs_sender_api: { API: { default: [Getter] }, Schema: { default: [Getter] } },
  s3_exporter: { Processor: { default: [Getter] }, Schema: { default: [Getter] } },
  s3_reader: {
    Fetcher: { default: [Getter] },
    Schema: { default: [Getter] },
    Slicer: { default: [Getter] }
  },
  s3_reader_api: { API: { default: [Getter] }, Schema: { default: [Getter] } },
  s3_sender_api: { API: { default: [Getter] }, Schema: { default: [Getter] } }
}
```

**Review by peterdemartini — APPROVED** [review]

**Pull request merged**